RCTAppleHealthKit/RCTAppleHealthKit.m

```diff
@@ -140,6 +140,12 @@ @implementation RCTAppleHealthKit
     [self fitness_getSamples:input callback:callback];
 }
 
+RCT_EXPORT_METHOD(getAnchoredWorkouts:(NSDictionary *)input callback:(RCTResponseSenderBlock)callback)
+{
+    [self _initializeHealthStore];
```

Comment:
Why is this method exported as well?

Reply:
And, right.It's just a difference of how our team uses our methods... By initializing the methods within the objective-c export, it allows us to call the method outside of `iniHealthKit`. So for example the `react-native-health` docs discuss using the library methods within AppleHealthKit.initHealthKit, like so:

```js
AppleHealthKit.initHealthKit(permissions, (error: string) => {
    AppleHealthKit.getAnchoredWorkouts()
}
```

Our team uses these functions through our codebase by calling the methods imperatively
`AppleHealthKit.getAnchoredWorkouts()`, like for example in a component's hook.

```js
useEffect(() => {
    AppleHealthKit.getAnchoredWorkouts()
},[])
```

We assume that healthkit permissions have been granted/denied. If they've been granted, the function will run and it won't if they've been denied.

However, I don't think that adding `self _initializeHealthStore` should affect users who write their functions within AppleHealthKit.initHealthKit, outside for making a redundant call to healthkit. Thoughts?

Reply:
That's a very smart approach to use and definitively simpler to follow and interact with

One addition to it, when the `AppleHealthKit.initHealthKit(...)` is called, iOS prompts the user with the Apple Health permissions if they are accessing the app for the first time. We could add an option to the method (something called `{ selfInitialize: true }`) to avoid enforcing that and changing the logic to existing apps that rely on that

We also can create another issue to replace all methods to accept this input, keeping their default values to `false`. On the next major release (v2) we make them all default to true and inform on the release notes.

What do you think? 🙂

Reply:
That looks good to me! Nice!

Reply:
@lucaspbordignon yeah, I think an optional param for all functions would be a nice addition!